compose/ChartboostMediationDemo/src/main/java/com/chartboost/mediation/sdk/demo/MainActivity.kt

```diff
@@ -0,0 +1,411 @@
+import androidx.compose.material3.SwitchDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.ContextCompat
+import com.chartboost.heliumsdk.ad.HeliumBannerAd
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.launch
+
+/**
+ * This is a demo app for the Chartboost Mediation SDK. It demonstrates how to use the SDK with Jetpack Compose.
+ * The app is written in Kotlin and uses the Compose UI toolkit.
+ *
+ * This Activity is the entry point of the app. It initializes the SDK and displays the UI.
+ *
+ * For ad loading and showing logic, reference the [AdController].
+ */
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Add the AdController as a lifecycle observer so that it can handle the lifecycle events
+        // of the activity.
+        lifecycle.addObserver(AdController)
+
+        setContent {
+            MaterialTheme {
+                Surface(color = MaterialTheme.colorScheme.background) {
+                    App()
+                }
+            }
+        }
+    }
+}
+
+/**
+ * The main UI of the app. It contains buttons to load and show interstitial and rewarded ads.
+ * Banner ads are loaded and shown automatically.
+ *
+ * Do any initialization and setup in here.
+ */
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun App() {
+    // Get the context to initialize the SDK, and to load and show ads.
+    val context = LocalContext.current
+
+    // A list of log messages to display in the log console.
+    val logState = remember { mutableStateListOf<String>() }
+
+    // Get the app ID, app signature, and placement names from the strings.xml file.
+    val appId = stringResource(R.string.app_id)
+    val appSignature = stringResource(R.string.app_signature)
+    val bannerPlacement = stringResource(R.string.banner_placement)
+    val interstitialPlacement = stringResource(R.string.interstitial_placement)
+    val rewardedAdPlacement = stringResource(R.string.rewarded_placement)
+
+    // Whether the show buttons for interstitial and rewarded ads should be enabled. They are
+    // disabled by default, and are enabled when the corresponding ad is loaded.
```

Comment:
Can we make it so that the show buttons are disabled after tapping show or when the ad is closed?